android/src/main/java/io/ably/flutter/plugin/AblyFlutterPlugin.java

```diff
@@ -28,22 +34,23 @@ public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBindin
     // them functionally equivalent. Only one of onAttachedToEngine or registerWith will be called
     // depending on the user's project. onAttachedToEngine or registerWith must both be defined
     // in the same class.
-    public static void registerWith(Registrar registrar) {
-        AblyFlutterPlugin.setupChannels(registrar.messenger());
+    public void registerWith(Registrar registrar) {
+        AblyFlutterPlugin.setupChannels(registrar.messenger(), applicationContext);
     }
 
-    private static void setupChannels(BinaryMessenger messenger) {
+    private static void setupChannels(BinaryMessenger messenger, Context applicationContext) {
         final MethodCodec codec = createCodec();
 
         final StreamsChannel streamsChannel = new StreamsChannel(messenger, "io.ably.flutter.stream", codec);
-        streamsChannel.setStreamHandlerFactory(arguments -> new AblyEventStreamHandler());
+        streamsChannel.setStreamHandlerFactory(arguments -> new AblyEventStreamHandler(applicationContext));
 
         final MethodChannel channel = new MethodChannel(messenger, "io.ably.flutter.plugin", codec);
         AblyMethodCallHandler methodCallHandler = AblyMethodCallHandler.getInstance(
                 channel,
                 // Streams channel will be reset on `register` method call
                 // and also on every hot-reload
-                streamsChannel::reset
+                streamsChannel::reset,
+            applicationContext
```

Comment:
Indentation mismatch with lines above.

Reply:
Fixed in 1cd73046d6db15fb8afad4d5882ba62469fa5c6a